sql/src/main/java/io/crate/operation/aggregation/impl/PercentileAggregation.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    public static class TDigestStateType extends DataType<TDigestState> implements Streamer<TDigestState>, DataTypeFactory {
+
+        private static final int ID = 5120;
+        public static final TDigestStateType INSTANCE = new TDigestStateType();
+
+        private TDigestStateType() {}
+
+        @Override
+        public DataType<?> create() {
+            return INSTANCE;
+        }
+
+        @Override
+        public TDigestState readValueFrom(StreamInput in) throws IOException {
+            return TDigestState.read(in);
+        }
+
+        @Override
+        public void writeValueTo(StreamOutput out, Object v) throws IOException {
+            TDigestState.write((TDigestState) v, out);
+        }
+
+        @Override
+        public int id() {
+            return ID;
+        }
+
+        @Override
+        public String getName() {
+            return "percentile_state";
+        }
+
+        @Override
+        public Streamer<?> streamer() {
+            return this;
+        }
+
+        @Override
+        public TDigestState value(Object value) throws IllegalArgumentException, ClassCastException {
+            return (TDigestState) value;
+        }
+
+        @Override
+        public int compareValueTo(TDigestState val1, TDigestState val2) {
+            return 0;
+        }
+    }
+
+    public static class TDigestState extends AVLTreeDigest {
+
+        private final double compression;
+        private final double[] fractions;
+
+        public TDigestState(double compression, double[] fractions) {
+            super(compression);
+            this.compression = compression;
+            this.fractions = fractions;
+        }
+
+        @Override
+        public double compression() {
+            return compression;
+        }
+
+        public double[] fractions() {
+            return fractions;
+        }
+
+        public static void write(TDigestState state, StreamOutput out) throws IOException {
+            out.writeDouble(state.compression);
+            out.writeDoubleArray(state.fractions);
+            out.writeVInt(state.centroidCount());
+            for (Centroid centroid : state.centroids()) {
+                out.writeDouble(centroid.mean());
+                out.writeVLong(centroid.count());
+            }
+        }
+
+        public static TDigestState read(StreamInput in) throws IOException {
+            double compression = in.readDouble();
+            double[] fractions = in.readDoubleArray();
+            TDigestState state = new TDigestState(compression, fractions);
+            int n = in.readVInt();
+            for (int i = 0; i < n; i++) {
+                state.add(in.readDouble(), in.readVInt());
+            }
+            return state;
+        }
+
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Nullable
+    @Override
+    public TDigestState newState(RamAccountingContext ramAccountingContext) {
+        return null;
```

Comment:
needs to return `null` since `fractions` attr in `TDigestState` cannot be modified later on in `iterate(...)`